ir/src/constraint_builder/integrity_constraints/mod.rs

```diff
@@ -25,10 +25,10 @@ impl ConstraintBuilder {
         match stmt {
             IntegrityStmt::Constraint(ConstraintType::Inline(constraint), _) => {
                 // add the left hand side expression to the graph.
-                let lhs = self.insert_expr(constraint.lhs())?;
+                let lhs = self.insert_expr(constraint.lhs().clone())?;
```

Comment:
Since we're changing to a consume pattern, we should break the constraint into parts at the top of this case, then the `clone` won't be necessary.

```
                let (lhs, rhs) = constraint.into_parts();
                // add the left hand side expression to the graph.
                let lhs = self.insert_expr(lhs)?;
```